boards/arm/nucleo_g031k8/doc/index.rst

```diff
@@ -0,0 +1,170 @@
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| CLOCK     | on-chip    | reset and clock control             |
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c controller                      |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi controller                      |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported in this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+``boards/arm/nucleo_g031k8/nucleo_g031k8_defconfig``
+
+Connections and IOs
+===================
+
+Each of the GPIO pins can be configured by software as output (push-pull or open-drain), as
+input (with or without pull-up or pull-down), or as peripheral alternate function. Most of the
+GPIO pins are shared with digital or analog alternate functions. All GPIOs are high current
+capable except for analog inputs.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- UART_2 TX/RX : PA2/PA3 (ST-Link Virtual Port Com)
+- I2C2 SCL/SDA : PA9/PA10 (Arduino I2C)
+- SPI1 SCK/MISO/MOSI : PB3/PB4/PB5 (Arduino SPI)
+- LD3       : PC6
+
+For mode details please refer to `STM32 Nucleo-32 board User Manual`_.
+
+Programming and Debugging
+*************************
+
+Applications for the ``nucleo_g031k8`` board configuration can be built and
+flashed in the usual way (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Flashing
+========
+
+Nucleo G031K8 board includes an ST-LINK/V2-1 embedded debug tool interface.
+This interface is supported by the openocd version included in the Zephyr SDK.
+
+Flashing an application to Nucleo G031K8
+----------------------------------------
+
+Here is an example for the :ref:`blinky-sample` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: nucleo_g031k8
+   :goals: build flash
+
+You will see the LED blinking every second.
+
+Debugging
+=========
+
+You can debug an application in the usual way.  Here is an example for the
+:ref:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: nucleo_g031k8
+   :maybe-skip-config:
+   :goals: debug
+
+Restriction
+***********
+
+On some boards, the board reset line is not used by the controller.
```

Comment:
Do you have more information on which board could demonstrate this behavior?

Reply:
From the two boards I bought, both have the option bit set to ignore the reset line.
The rest of my information comes from https://os.mbed.com/platforms/ST-Nucleo-G031K8/#restriction

Although I have a suspicion, that all boards are configured like that, I am not sure.
This is also the reason, for using the stlink-dap interface in the OpenOCD configuration, since this works in both cases.

Reply:
Thanks for the pointer. I'll try to get more info from there.